flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/FunctionGenerator.scala

```diff
@@ -146,12 +146,25 @@ object FunctionGenerator {
     STRING_TYPE_INFO,
     BuiltInMethod.OVERLAY.method)
 
+  addSqlFunctionMethod(
+    ASCII,
+    Seq(STRING_TYPE_INFO),
+    STRING_TYPE_INFO,
```

Comment:
Should it return int type instead?

Reply:
I know that in normal scenarios, we should return an int, but we need to handle the input exception properly. We need to return null in the case where the input is null, and return "" if the input is "". If we return the int type directly, then we have difficulty dealing with these two exception scenarios. So I chose to return a string representation of the numeric type.

Reply:
got it, but why do we just returning null for empty string input? I imagine someone might need to CAST(ASCII(argument) AS INTEGER) for their use, the empty string output will cause exception.

Reply:
@suez1224 thanks for your reminder. You are right, In the scenario you mentioned, it throws an exception. I found that we don't need to check if the input of the function is null. It seems that the Flink framework has been processed by default. All the input is null and will return null (man can not interfere). Then we only need to deal with "", in this case, the ASCII code is 0. So, I directly changed the return value of this method to the Int type. 

cc @hequn8128 : 

It seems that we don't need to consider the case where the input is null. Once the input of the function is null, this function will not be called at all. I debugged it through the test class. It looks like this, you can confirm it.

Reply:
Hi @yanghua . Yes, currently,  `functionCodeGenerator` in `DataStreamCalc` will process nullinput. But I think it is good to make the function capable of processing nullinput. What do you think?

Reply:
@hequn8128 yes, I agree this point.